kore/src/Kore/Parser/Lexer.x

```diff
@@ -0,0 +1,328 @@
+
+{- | Data type for the raw lexical data of a Token. Essentially an enumeration
+specifying which token a particular Token represents. Additionally contains the
+semantic data associated with identifier and string tokens.
+-}
+data TokenClass
+  = TokenModule
+  | TokenEndModule
+  | TokenImport
+  | TokenSort
+  | TokenSymbol
+  | TokenWhere
+  | TokenAlias
+  | TokenAxiom
+  | TokenClaim
+  | TokenHookedSort
+  | TokenHookedSymbol
+  | TokenColonEqual
+  | TokenColon
+  | TokenLeftBrace
+  | TokenRightBrace
+  | TokenLeftBracket
+  | TokenRightBracket
+  | TokenLeftParen
+  | TokenRightParen
+  | TokenComma
+  | TokenTop
+  | TokenBottom
+  | TokenNot
+  | TokenAnd
+  | TokenOr
+  | TokenImplies
+  | TokenIff
+  | TokenExists
+  | TokenForall
+  | TokenMu
+  | TokenNu
+  | TokenCeil
+  | TokenFloor
+  | TokenEquals
+  | TokenIn
+  | TokenNext
+  | TokenRewrites
+  | TokenDomainValue
+  | TokenLeftAssoc
+  | TokenRightAssoc
+  | TokenIdent Text
+  | TokenSetIdent Text
+  | TokenString Text
+  | TokenEOF
+  deriving stock (Eq, Show)
+
+-- -----------------------------------------------------------------------------
+-- Alex wrapper code.
+--
+-- Taken and modified from public domain code that is part of Alex:
+-- https://github.com/haskell/alex/blob/3.2.6/templates/wrappers.hs
+
+
+alexScanUser :: () -> AlexInput -> Int -> AlexReturn (AlexInput -> Int -> Alex Token)
+
+data AlexInput = AlexInput { alexPosn :: {-# UNPACK #-} !AlexPosn,
+                             alexChar :: {-# UNPACK #-} !Char,
+                             alexStr :: {-# UNPACK #-} !ByteString.ByteString,
+                             alexBytePos :: {-# UNPACK #-} !Int}
+
+data AlexPosn = AlexPn !FilePath !Int !Int !Int
+        deriving (Eq,Show)
+
+data AlexState = AlexState {
+        alex_pos :: !AlexPosn,  -- position at current input location
+        alex_bpos:: !Int,     -- bytes consumed so far
+        alex_inp :: ByteString.ByteString,      -- the current input
+        alex_chr :: !Char,      -- the character before the input
+        alex_scd :: !Int        -- the current startcode
+    }
+
+newtype Alex a = Alex { unAlex :: AlexState -> Either String (AlexState, a) }
+
+instance Functor Alex where
+  fmap f a = Alex $ \s -> case unAlex a s of
+                            Left msg -> Left msg
+                            Right (s', a') -> Right (s', f a')
+
+instance Applicative Alex where
+  pure a   = Alex $ \s -> Right (s, a)
+  fa <*> a = Alex $ \s -> case unAlex fa s of
+                            Left msg -> Left msg
+                            Right (s', f) -> case unAlex a s' of
+                                               Left msg -> Left msg
+                                               Right (s'', b) -> Right (s'', f b)
+
+instance Monad Alex where
+  m >>= k  = Alex $ \s -> case unAlex m s of
+                                Left msg -> Left msg
+                                Right (s',a) -> unAlex (k a) s'
+  return = App.pure
```

Comment:
```suggestion
```
I believe we only support GHC versions that no longer require a definition of `return`.

If I'm wrong about that, this definition should also have an `INLINE` pragma.